libibumad/umad.h

```diff
@@ -54,6 +54,21 @@ typedef __be16 __attribute__((deprecated)) be16_t;
 typedef __be32 __attribute__((deprecated)) be32_t;
 typedef __be64 __attribute__((deprecated)) be64_t;
 
+typedef union _ib_gid
+{
+	uint8_t	raw[16];
+
+	struct {
+		__be64	prefix;
+		__be64	interface_id;
+	} unicast;
+
+	struct {
+		uint8_t	header[2];
+		uint8_t	raw_group_id[14];
+	} multicast;
+} ib_gid_t;
```

Comment:
Bart,

That sounds like a really good plan...  libibumad and others should migrate to that definition.  I'll put it on my list for the diags and libibmad.  They don't really use ib_gid_t that much though.

I still think opensm should stop exporting this as well.  Hal do you agree?  There are probably other types which could be migrated to the core as well.  I just don't have time right now to do an audit.